src/WinRT.Interop.Generator/Generation/InteropGenerator.Discover.cs

```diff
@@ -404,28 +404,26 @@ private static void ValidateWinRTRuntimeDllVersion2References(InteropGeneratorAr
     /// <summary>
     /// Creates an <see cref="InteropReferences"/> instance that can be used for the discovery phase.
     /// </summary>
-    /// <param name="corLibTypeFactory">The <see cref="CorLibTypeFactory"/> currently in use.</param>
+    /// <param name="module">The module currently being analyzed.</param>
     /// <returns>The <see cref="InteropReferences"/> instance to use for the discovery phase.</returns>
     [SuppressMessage("Style", "IDE0059", Justification = "Creating the 'AssemblyDefinition'-s is used to bind them to the contained modules.")]
-    private static InteropReferences CreateDiscoveryInteropReferences(CorLibTypeFactory corLibTypeFactory)
+    private static InteropReferences CreateDiscoveryInteropReferences(ModuleDefinition module)
     {
         // Create the interop references scoped to this module, which we need to lookup some references from
         // the 'WinRT.Runtime.dll' assembly. We haven't loaded it just here here, so we can't use the real
         // module definition for it. Instead, we just create an empty one here. This is only used to create
         // type and member references to APIs defined in that module, so this is good enough for this scenario.
         // We also do the same for the Windows Runtime projection assembly, the exact version doesn't matter.
         Version windowsRuntimeVersion = Assembly.GetExecutingAssembly().GetName().Version ?? new Version(0, 0, 0, 0);
-        ModuleDefinition windowsRuntimeModule = new("WinRT.Runtime2.dll"u8, KnownCorLibs.SystemRuntime_v10_0_0_0);
-        AssemblyDefinition windowsRuntimeAssembly = new("WinRT.Runtime2", windowsRuntimeVersion) { Modules = { windowsRuntimeModule } };
-        ModuleDefinition windowsSdkProjectionModule = new("Microsoft.Windows.SDK.NET.dll"u8, KnownCorLibs.SystemRuntime_v10_0_0_0);
-        AssemblyDefinition windowsSdkProjectionAssembly = new("Microsoft.Windows.SDK.NET", new Version(10, 0, 0, 0)) { Modules = { windowsSdkProjectionModule } };
+        AssemblyReference windowsRuntimeAssembly = new("WinRT.Runtime"u8, new Version(10, 0, 0, 0));
```

Comment:
10.0?

Reply:
Oops, copy paste error, fixed!

Reply:
Well all Windows SDK .dll-s are version 10.0.XXXXX.0, so using 10.0 is fine. This is just used as a reference assembly basically, so it'll always bind to the same .dll with a newer version that's compatible. Like, I've been testing this code locally and it works just fine, with a Windows SDK .dll from like the latest 26100 package.